### ISSUE TITLE
Multiple issues with commit c87d93a

### ISSUE THREAD

**Issue by biork** [Illumina/DRAGMAP#27]
I have been rerunning my torture test since issues I posted earlier have apparently been addressed.
The test:
1. generates random genomes and simulates reads from those genomes parameterizable in many ways
2. runs dragen-os to build the hashmaps
3. runs dragen-os to map the simulated reads
4. verifies the results

The dragen-os executable passes the test more often than not, but it fails roughly 10% of the time in one of two ways:
1. infinite loops in the hashtable building phase
2. a reliable crash from a macro assertion always at lib/reference/Hashtable.cpp:96 (even in release builds)
FWIW, the random genomes are small, and unrealistic at least in that sense, but they're not pathologically small and I see both failures for a range of simulation parameters. The hangs and crashes both seem to be more common in simulated genomes with high (hundreds) of contigs rather than 10's, but I have not tried to quantify this.

**Comment by rizkg** [comment]
Hi, and thanks for the update.
Just to make sure, the crashes you experience are always in the hash building phase, not when doing mapping ?

**Comment by biork** [comment]
No, the crashes (at Hashtable.cpp:96) are always in the mapping phase, and the (apparent) infinite loop is always in the hash building phase.

**Comment by biork** [comment]
Hi Guillaume, sorry for delay. I have a 47M archive I could upload somewhere containing one of multiple example runs that I just confirmed aborts at the Hashtable.cpp:96 assertion.
